lib/engine/game/g_18_mag.rb

```diff
@@ -40,6 +42,38 @@ def setup
           hex.tile.cities[0].place_token(@mine, @mine.next_token)
         end
         @mine.tokens.each { |t| t.type = :neutral }
+
+        # IPO and float all corporations with semi-randomly chosen prices
+        # They will start off in receivership with all shares in market
+        prices = START_PRICES.dup
+        rand_prices = @corporations.size.times.map do |_|
+          prices.rotate!(rand % prices.size)
+          prices.pop
```

Comment:
what is this trying to do? could you just do START_PRICES.sort_by { rand }

Reply:
Yeah, I figured there was an elegant way of during this, but I just brute-forced it (and not very efficiently either).